server/internal/urn/tools.go

```diff
@@ -0,0 +1,170 @@
+package urn
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/speakeasy-api/gram/server/internal/constants"
+)
+
+// maxURNSegmentLength is the maximum length of each segment in tool URNs (kind,
+// source, name). Application code can further enforce shorter lengths. The
+// purpose of this is to place an upper bound on parsing logic.
+const maxToolSegmentLength = 100
+
+const delimiter = ":"
+
+var ErrInvalidTool = errors.New("invalid tool urn")
```

Comment:
```suggestion
var ErrInvalidURN= errors.New("invalid tool urn")
```

Reply:
I'm not a fan of package exports that stutter with the package name. Changing to `ErrInvalid`.